### PR TITLE
DOC: add DOI to references

### PR DESCRIPTION
**Pull request by sciunto**
From the mailing list, we are convinced of the benefits from DOIs.

I started with thresholding since I worked recently on this file. I would be cool to propagate this policy to existing references (tedious) and new PR.

### CODE REVIEW

File: skimage/filters/thresholding.py [hunk 1]
```diff
@@ -330,10 +330,10 @@ def threshold_yen(image, nbins=256):
     ----------
     .. [1] Yen J.C., Chang F.J., and Chang S. (1995) "A New Criterion
            for Automatic Multilevel Thresholding" IEEE Trans. on Image
-           Processing, 4(3): 370-378
+           Processing, 4(3): 370-378. DOI:10.1109/83.366472
```
Comment (soupault) [inline skimage/filters/thresholding.py]: Hmm.. Here and below there is a syntax inconsistence: `.`, `,`, ` ` before the DOI.

Comment (sciunto) [inline skimage/filters/thresholding.py]: The problem is that all citations have inconsistent formats. As it's not generated from a bibtex for ex, It's a tough work.

**Comment by jni** [comment]
👍 from me

**Comment by soupault** [comment]
Thanks @sciunto!

**Pull request merged**